packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -151,23 +188,17 @@ export class PendingStateManager implements IDisposable {
             opMetadata,
         };
 
-        this.pendingStates.push(pendingMessage);
-
-        this._pendingMessagesCount++;
+        this.pendingMessages.push(pendingMessage);
     }
 
     /**
-     * Called when flush() is called on the ContainerRuntime to manually flush messages.
+     * @deprecated Use batch metadata to indicate end of batch. To be removed in 2.0.0-internal.4.0.0 (AB#2496)
```

Comment:
Why do we need back-compat here? Aren't all the call sites in container runtime which is in the same layer? Is it because the change to add batch metadata to messages is behind a feature flag?

Reply:
Andre mentioned this function was only deprecated and not removed (before, I made this method do nothing at all). So, I made this intermediary step if somebody still wants to use the `onFlush` method.

Looking at it more, this class isn't exported. So, it's probably fine if this method loses its functionality as no one can extend it?

Reply:
@agarwal-navin FYI, I don't think adding the batch metadata is part of a feature flag? There isn't any check in `ContainerRuntime.flushBatch(...)`